src/Extensions/ElementalSubsiteExtension.php

```diff
@@ -32,6 +32,17 @@ public function updateCMSFields(FieldList $fields)
         }
     }
 
+    /**
+     * Ensure the new block inherits the current subsite id
+     *
+     * @return void
+     */
+    public function onBeforeWrite()
+    {
+        if (class_exists(Subsite::class)) {
+            $this->owner->SubsiteID = Subsitestate::singleton()->getSubsiteId();
```

Comment:
Looks good, but can we match case here? e.g.  `SubsiteState`

Reply:
@adrexia good spot, updated!